### DISCUSSION TITLE
Colour filled - event

### DISCUSSION THREAD

**Discussion by ghost** [ebarellar/LFColoringBook#12] [Q&A]
![Simulator Screen Shot - iPhone 11 - 2022-01-01 at 02 27 15](https://user-images.githubusercontent.com/76158179/147838819-1e75f1ab-7c9c-400a-89f2-0c9287a6b166.png)

I stuck in one issue that, when I finish colouring in my art after that I want to display popup like painting has been completed so can you please let me know that by using which method or condition I can easily get callback when I finish colouring in design.

Thanks in advance.

**Comment by ebarellar** [comment]
You can probably do that whenever the delegate?.viewDidEndDrawing() method is called and then apply some logic to check if the drawing is "complete"

My guess is that you could check which mask was being processed, read the pixels from that region and check if all have alpha > 0. Then keep track on which masks are "full" and send the warning once all are done. I hope the performance hit isn't that bad

**Comment by ghost** [comment]
Thank you for your reply.

Can you please tell me how to check which mask is currently being processed? Because I tried it using for loop in viewDidEndDrawing() method but it slow down my application. Please guide me with that as my project is on hold because of that thing.
